install/kubernetes/cilium/templates/clustermesh-apiserver/servicemonitor.yaml

```diff
@@ -0,0 +1,57 @@
+{{- if and
+  (or .Values.externalWorkloads.enabled .Values.clustermesh.useAPIServer)
+  (or .Values.clustermesh.apiserver.metrics.enabled .Values.clustermesh.apiserver.metrics.etcd.enabled)
+  .Values.clustermesh.apiserver.metrics.serviceMonitor.enabled }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: clustermesh-apiserver
+  namespace: {{ .Values.clustermesh.apiserver.metrics.serviceMonitor.namespace | default .Release.Namespace }}
+  labels:
+    app.kubernetes.io/part-of: cilium
+    {{- with .Values.clustermesh.apiserver.metrics.serviceMonitor.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- with .Values.clustermesh.apiserver.metrics.serviceMonitor.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: clustermesh-apiserver
+      app.kubernetes.io/component: metrics
+  namespaceSelector:
+    matchNames:
+    - {{ .Release.Namespace }}
+  endpoints:
+  {{- if .Values.clustermesh.apiserver.metrics.enabled }}
+  - port: apiserv-metrics
```

Comment:
Any reason why these port names have a shorter form `apiserv-metrics` and  not `apiserver-metrics`?

Reply:
Yes, port names are limited to 15 characters to comply with IANA_SVC_NAME constrains: [RFC6335](https://www.rfc-archive.org/getrfc.php?rfc=6335#gsc.tab=0).